### ISSUE TITLE
Hello, I can't find where build_network is.

### ISSUE THREAD

**Issue by zhangjun221** [longbai1006/EndoUIC#1]
![1721100136333](https://github.com/user-attachments/assets/949205e8-7cd5-4453-9269-fb5083fa70bc)

**Comment by yangh0119** [comment]
Have you solved it？I meet the same problem。

**Comment by longbai1006** [comment]
Hi! Did you compile basicsr correctly? You may find the detailed compiling method at https://github.com/limuloo/PyDIff (should be the same in our README).

**Comment by zhangjun221** [comment]
> Hi! Did you compile basicsr correctly? You may find the detailed compiling method at https://github.com/limuloo/PyDIff (should be the same in our README).你好！你正确编译了 basicsr 吗？详细的编译方法可以在https://github.com/limuloo/PyDIff找到（应该和我们的README一样）。
![273b68b0d86e83f8be47a0b0eae90ba](https://github.com/user-attachments/assets/2b2ac611-d817-4f12-b9b7-7d1e65afa084)

I compiled according to this environment, but I couldn't find the build_network function. I also don't see this function in the code. Is there something else that hasn't been compiled yet?

**Comment by longbai1006** [comment]
I think I missed this file in my repo. I will update it ASAP.
https://github.com/limuloo/PyDIff/blob/99512399045a9439a34c76f4d5aab2248a5e0890/BasicSR-light/basicsr/archs/__init__.py#L19

**Comment by zhangjun221** [comment]
> I think I missed this file in my repo. I will update it ASAP. https://github.com/limuloo/PyDIff/blob/99512399045a9439a34c76f4d5aab2248a5e0890/BasicSR-light/basicsr/archs/__init__.py#L19

Alright, thank you very much for your response. I would like to know if the mamba_ssm library is needed. I didn't see it in your environment.txt, but it seems that your model calls functions from the mamba_ssm library.

**Comment by yangh0119** [comment]
> 你好！你编译 basicsr 正确了吗？您可以在 https://github.com/limuloo/PyDIff 找到详细的编译方法（在我们的 README 中应该相同）。

I was not successful in compiling Basicsr. I installed it using pip, but when running the code, a new error appeared, indicating a version mismatch

**Comment by longbai1006** [comment]
> > 你好！你编译 basicsr 正确了吗？您可以在 https://github.com/limuloo/PyDIff 找到详细的编译方法（在我们的 README 中应该相同）。
> 
> I was not successful in compiling Basicsr. I installed it using pip, but when running the code, a new error appeared, indicating a version mismatch

You may try compile it by the code from https://github.com/limuloo/PyDIff

**Comment by longbai1006** [comment]
> > I think I missed this file in my repo. I will update it ASAP. https://github.com/limuloo/PyDIff/blob/99512399045a9439a34c76f4d5aab2248a5e0890/BasicSR-light/basicsr/archs/__init__.py#L19
> 
> Alright, thank you very much for your response. I would like to know if the mamba_ssm library is needed. I didn't see it in your environment.txt, but it seems that your model calls functions from the mamba_ssm library.

Yes. Firstly, you may try to compile basicsr by https://github.com/limuloo/PyDIff
Then, for mamba installation, you can refer to this code https://github.com/state-spaces/mamba/issues/55. I am working on organizing this part of code into our README.

**Comment by zhangjun221** [comment]
> > > I think I missed this file in my repo. I will update it ASAP. https://github.com/limuloo/PyDIff/blob/99512399045a9439a34c76f4d5aab2248a5e0890/BasicSR-light/basicsr/archs/__init__.py#L19
> > 
> > 
> > Alright, thank you very much for your response. I would like to know if the mamba_ssm library is needed. I didn't see it in your environment.txt, but it seems that your model calls functions from the mamba_ssm library.
> 
> Yes. Firstly, you may try to compile basicsr by https://github.com/limuloo/PyDIff Then, for mamba installation, you can refer to this code [state-spaces/mamba#55](https://github.com/state-spaces/mamba/issues/55). I am working on organizing this part of code into our README.

I have successfully installed Mamba and compiled it, but build_network is a function I wrote myself, and I'm not sure if it is correct.

**Comment by longbai1006** [comment]
I also put one here: https://github.com/longbai1006/EndoUIC/blob/4eec14e0a3d4dbf91c72ce12aa34b73b31443258/BasicSR-light/basicsr/archs/__init__.py#L19

**Comment by zhangjun221** [comment]
> I also put one here:
> 
> https://github.com/longbai1006/EndoUIC/blob/4eec14e0a3d4dbf91c72ce12aa34b73b31443258/BasicSR-light/basicsr/archs/__init__.py#L19

Thank you very much, and I wish you success in your work.
